pandas/tests/extension/test_numpy.py

```diff
@@ -78,7 +78,7 @@ def data_for_sorting(allow_in_pandas, dtype):
     if dtype.numpy_dtype == "object":
         # Use an empty tuple for first element, then remove,
         # to disable np.array's shape inference.
-        return PandasArray(np.array([(), (2,), (3,), (1,)])[1:])
+        return PandasArray(np.array([(), (2,), (3,), (1,)], dtype=object)[1:])
```

Comment:
@TomAugspurger does the comment above about shape inference have any bearing on the inference here?

Reply:
I think using `object` dtype is still correct, and I think the comment is still correct.

If we want to get rid of the empty tuple / shape stuff, we could probably allocate the array with the right shape and set the values later

```python
In [6]: a = np.empty((3,), dtype=object)

In [7]: a[:] = (2,), (3,), (1,)
```